### DISCUSSION TITLE
Allow getServerSideProps to be exported

### DISCUSSION THREAD

**Discussion by mrclay** [vercel/next.js#17960] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

Our app is dual-hosted: In an intranet it shows fresh content from a CMS that editors are working in. However the final product is required to be a static build with the content frozen at build-time.

The final site can't call the CMS, so `getInitialProps` is out. And `getServerSideProps` is--seemingly arbitrarily--banned from export. So we're limping along with `getStaticProps` and `revalidate: 1`. So CMS editors see fresh content on the very next refresh, we have to ignore the props from Next and re-fetch from the client.

## Describe the solution you'd like

`getServerSideProps` could export just like `getStaticProps`, freezing output into JSON files.

**Comment by timneutkens** [comment]
Preview mode exists exactly for this use case: https://nextjs.org/docs/advanced-features/preview-mode

**Reply by mrclay** [reply]
Our whole site will require preview mode on. Is there a declarative way to set this or do I have to make sure every browser gets redirected through the API endpoint to get the cookie?

**Reply by timneutkens** [reply]
You'll have to redirect them through an API endpoint to get the cookie indeed.
